README.rst

```diff
@@ -223,11 +359,97 @@ Configure celery (you may use a different broker if you want):
+        option management_interface 'tun0'
+
+The different modules of OpenWISP use the *management interface* for providing the following features:
+
+- `Pushing configuration changes <https://github.com/openwisp/openwisp-controller#how-to-configure-push-updates>`_
+  from OpenWISP to the device.
+- `Executing commands remotely on the device <https://github.com/openwisp/openwisp-controller#sending-commands-to-devices>`_.
+- `Running checks on the device <https://github.com/openwisp/openwisp-monitoring#available-checks>`_.
+- `Performing firmware upgrades on the device <https://github.com/openwisp/openwisp-firmware-upgrader#perform-a-firmware-upgrade-to-a-specific-device>`_
+
+It is **required** to set up the *management interface* for functioning
+of above-listed features.
+
+You can only get away from setting up a management interface
+if OpenWISP is deployed in a Layer 2 network, i.e. OpenWISP can reach the devices
+on their **Last IP**. In this scenario, set the `"OPENWISP_MONITORING_MANAGEMENT_IP_ONLY" <#openwisp-monitoring-management-ip-only>`_
+setting in OpenWISP to ``False``.
+
+Install monitoring packages on the device
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Follow the `instructions from openwrt-openwisp-monitoring documentation <https://github.com/openwisp/openwrt-openwisp-monitoring/tree/master#install-pre-compiled-packages>`_
+to install ``openwisp_monitoring`` and ``netjson_monitoring``
+packages on your device. These packages collect and send the
+required data from the device and are required for the functioning
+of `checks <#available-checks>`_ and `metrics <#openwisp_monitoring_metrics>`_
+features.
+
+**Note**: If you are an existing user of *openwisp-monitoring* and are using
+a *monitoring template* for collecting metrics, you should follow the
+instructions in `Migrating from monitoring scripts to monitoring packages <#migrating-from-monitoring-scripts-to-monitoring-packages>`_
+section of this documentation.
+
+Metric collection
+~~~~~~~~~~~~~~~~~
+
+*openwisp-monitoring* collects `different metrics of the device <https://github.com/openwisp/openwisp-monitoring#default-metrics>`_.
+Some of these metrics are collected by OpenWISP itself through means of **active checks**
+while others are sent to OpenWISP passively by the `openwisp-monitoring agent <#install-monitoring-packages-on-the-device>`_
+installed on the device.
+
+The `"Available Checks" <#available-checks>`_ section of this documentation lists
+**active checks** implemented in openwisp-monitoring.
+
+Creating checks for a device
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+By default, *openwisp-monitoring* creates `active checks <#available-checks>`_
+for all devices. These checks are created and executed asynchronously.
+
+If you don't see checks created immediately after a device registers, don't
+be alarmed. It may take some time for the asynchronous worker(s) to create
+checks for the device. Unless you have disabled the automatic creation of a check.
```

Comment:
I think last sentence is not entirely correct, it shouldn't take long, a few seconds max, the user may be misled by this and waste time if there's any issue.